tests/tests/tx_gossip.rs

```diff
@@ -90,3 +96,129 @@ async fn test_tx_gossiping() {
+                        .with_name(format!("{pub_key}:{i}"))
+                        .bootstrap_type(bootstrap_type)
+                        // Validator wants to send invalid transactions.
+                        .utxo_validation(false),
+                )
+            })
+        }),
+    )
+    .await;
+
+    let authority = &producers[0];
+    let sentry = &validators[0];
+
+    // Time for nodes to connect to each other.
+    tokio::time::sleep(Duration::from_secs(2)).await;
+
+    use rand::Rng;
+
+    for _ in 0..NUMBER_OF_INVALID_TXS {
+        let invalid_tx = TransactionBuilder::script(vec![], vec![])
+            .add_unsigned_coin_input(
+                rng.gen(),
+                rng.gen(),
+                rng.gen(),
+                rng.gen(),
+                Default::default(),
+                Default::default(),
+            )
+            .finalize()
+            .into();
+
+        sentry.node.submit(invalid_tx).await.expect(
+            "Should accept invalid transaction because `utxo_validation = false`.",
+        );
+    }
+
+    // Give some time to receive all invalid transactions.
+    tokio::time::sleep(Duration::from_secs(5)).await;
+
+    let mut authority_blocks = authority.node.shared.block_importer.block_stream();
+
+    // Submit a valid transaction from banned sentry to an authority node.
+    let valid_transaction = authority.test_transactions()[0].clone();
+    sentry
+        .node
+        .submit(valid_transaction.clone())
+        .await
+        .expect("Transaction is valid");
+    let _ = tokio::time::timeout(Duration::from_secs(5), authority_blocks.next()).await;
+
+    let authority_client = FuelClient::from(authority.node.bound_address);
+    authority_client
+        .transaction_status(&valid_transaction.id(&Default::default()))
+        .await
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_tx_gossiping_reserved_nodes_invalid_txs() {
+    // Test verifies that gossiping of invalid transactions from reserved
+    // nodes doesn't decrease its reputation.
+    // After gossiping `NUMBER_OF_INVALID_TXS` transactions,
+    // we will gossip one valid transaction, and it should be included.
+    let status = test_tx_gossiping_invalid_txs(BootstrapType::ReservedNodes).await;
+
+    assert!(matches!(
+        status,
+        Ok(fuel_core_client::client::types::TransactionStatus::Success { .. })
+    ));
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_tx_gossiping_non_reserved_nodes_invalid_txs() {
+    // This test is opposite to the `test_tx_gossiping_reserved_nodes_invalid_txs`.
+    // It verifies that gossiping about invalid transactions from
+    // non-reserved peers disconnects them.
+    //
+    // The test sends `NUMBER_OF_INVALID_TXS` invalid transactions,
+    // and verifies that sending a valid one will be ignored.
+    let status = test_tx_gossiping_invalid_txs(BootstrapType::BootstrapNodes).await;
+
+    assert!(status.is_err());
```

Comment:
Is it possible to assert a match for the specific error here, so we know that this error occurs because the sender was disconnected/rejected?